src/lib/output/ResultsProcessor.ts

```diff
@@ -0,0 +1,25 @@
+import {Results} from "./Results";
+
+/**
+ * Interface to process run results
+ */
+export interface ResultsProcessor {
+	processResults(results: Results): Promise<void>;
+}
+
+/**
+ * A composite results processor
+ */
+export class CompositeResultsProcessor implements ResultsProcessor {
+	private readonly delegates: ResultsProcessor[];
+
+	public constructor(delegateResultsProcessors: ResultsProcessor[]) {
+		this.delegates = delegateResultsProcessors;
+	}
+
+	async processResults(results: Results): Promise<void> {
+		for (const delegate of this.delegates) {
+			await delegate.processResults(results);
+		}
+	}
```

Comment:
I was originally going to simply use the Promise.all(...) here and just join all the promises under a single promise. But this wasn't so easy because it converts the final promise to a Promise<void[]> instead of a Promise<void> so i would have to create a custom Promise to do this. But then I got thinking that we probably want this composite to be syncronous and execute each delegate in order. For example if we create a SeverityCheckResultsProcessor ... we would want that last in order to throw an error after all other results processors were finished. And this code looks cleaner anyway.

Reply:
Are you positive that's how `Promise.all()` works? We're using `Promise.all()` in `DefaultRuleManager.ts` to get an array of results (see line 161), so it seems like we could do what you're describing.
Also, I'm not sure I understand why it's beneficial to have them run synchronously. Can you elaborate on that?

Reply:
The composite pattern here requires that the composite acts as a single instance that adheres to the interface. The interface wants Promise<void>. Now I could switch it to Promise<any> or something to that effect.

Synchronous execution is what we have today... but also it allows us to have order enforcement. So if we want to have an output processor and then an exception (severity check) based processor... if we didn't run them in order then we could end up erroring out before the file is created... which wouldn't be what we wanted.